homeassistant/components/homeassistant_analytics/coordinator.py

```diff
@@ -0,0 +1,50 @@
+"""DataUpdateCoordinator for the Homeassistant Analytics integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+
+from python_homeassistant_analytics import (
+    HomeassistantAnalyticsClient,
+    HomeassistantAnalyticsConnectionError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_TRACKED_INTEGRATIONS, DOMAIN, LOGGER
+
+
+class HomeassistantAnalyticsDataUpdateCoordinator(
+    DataUpdateCoordinator[dict[str, int]]
+):
+    """A Homeassistant Analytics Data Update Coordinator."""
+
+    config_entry: ConfigEntry
+
+    def __init__(
+        self, hass: HomeAssistant, client: HomeassistantAnalyticsClient
+    ) -> None:
+        """Initialize the Homeassistant Analytics data coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(hours=12),
+        )
+        self._client = client
```

Comment:
why do we pass the client and don't init it in here?

Reply:
No real reason